packages/sdk/schema/src/util.ts

```diff
@@ -80,3 +80,23 @@ const toFieldValueType = (type: AST.AST): { format?: FormatEnum; type: TypeEnum
   // TODO(burdon): Better fallback?
   return { type: TypeEnum.String, format: FormatEnum.JSON };
 };
+
+/**
+ * Creates or updates echo annotations for SingleSelect options in a JSON Schema property.
+ */
+export const makeSingleSelectAnnotations = (
+  jsonProperty: JsonSchemaType,
+  options: Array<{ id: string; title?: string; color?: string }>,
+) => {
+  jsonProperty.enum = options.map(({ id }) => id);
+  jsonProperty.format = FormatEnum.SingleSelect;
+  jsonProperty.echo = {
+    annotations: {
+      singleSelect: {
```

Comment:
we're we going to call these `oneOf`?

Reply:
@dmaretskyi and I decided on a new format which would play better with the effect-json serializer and not cause us the issues we've been seeing.

I think `singleSelect` inside the annotation works better because JsonSchema still has it's own `oneOf` concept and I'd like to not mix the two. I'm happy to change it over if you guys think it's worth it.